bulk-model-sync-lib/src/jvmTest/kotlin/org/modelix/model/sync/bulk/ModelImporterOrderPropertyTest.kt

```diff
@@ -0,0 +1,508 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.sync.bulk
+
+import org.junit.jupiter.api.Test
+import org.modelix.model.api.IBranch
+import org.modelix.model.api.INode
+import org.modelix.model.api.PBranch
+import org.modelix.model.api.SimpleChildLink
+import org.modelix.model.api.SimpleConcept
+import org.modelix.model.api.SimpleLanguage
+import org.modelix.model.api.SimpleProperty
+import org.modelix.model.api.getRootNode
+import org.modelix.model.client.IdGenerator
+import org.modelix.model.data.ModelData
+import org.modelix.model.lazy.CLTree
+import org.modelix.model.lazy.ObjectStoreCache
+import org.modelix.model.persistent.MapBasedStore
+import org.modelix.model.withAutoTransactions
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+/**
+ * Tests the parts of the import algorithm,
+ * that checks whether is a child node is in an ordered or unordered role;
+ * and that handles the importing children with of unordered roles.
+ */
+class ModelImporterOrderPropertyTest {
+    object TestLanguage : SimpleLanguage("TestLanguage", uid = "testLanguage") {
+
+        override var includedConcepts = arrayOf<SimpleConcept>(AConcept)
+
+        object AConcept : SimpleConcept(
+            conceptName = "aConcept",
+            uid = "uid1",
+        ) {
+            init {
+                addConcept(this)
+                // The child links in this concept are special, because they are unordered.
+                SimpleChildLink(
+                    simpleName = "aChildLink1",
+                    isMultiple = true,
+                    isOptional = true,
+                    isOrdered = false,
+                    targetConcept = AConcept,
+                    uid = "uid2",
+                ).also(this::addChildLink)
+                SimpleChildLink(
+                    simpleName = "aChildLink2",
+                    isMultiple = true,
+                    isOptional = true,
+                    isOrdered = false,
+                    targetConcept = AConcept,
+                    uid = "uid3",
+                ).also(this::addChildLink)
+            }
+        }
+    }
+
+    init {
+        TestLanguage.register()
+    }
+
+    private fun loadModelIntoBranch(modelData: ModelData): IBranch {
+        val store = ObjectStoreCache(MapBasedStore())
+        val tree = CLTree(
+            data = null,
+            repositoryId_ = null,
+            store_ = store,
+            useRoleIds = true,
+        )
+        val idGenerator = IdGenerator.getInstance(1)
+        val pBranch = PBranch(tree, idGenerator)
+
+        pBranch.runWrite {
+            modelData.load(pBranch)
+        }
+
+        return pBranch.withAutoTransactions()
+    }
+
+    @Test
+    fun `unordered children are not reorder in import`() {
```

Comment:
```suggestion
    fun `unordered children are not reorder`() {
```

I think the import context is already clear from the test class and that way I don't have to parse the somewhat complex grammar of the test method names ;-)